rasa/core/migrator.py

```diff
@@ -0,0 +1,214 @@
+            )
+
+        if not self.requested_conversation_ids:
+            return conversation_ids_in_tracker_store
+
+        missing_ids_in_tracker_store = (
+            set(self.requested_conversation_ids) - conversation_ids_in_tracker_store
+        )
+
+        if missing_ids_in_tracker_store:
+            cli_utils.print_warning(
+                f"Could not find the following requested "
+                f"conversation IDs in connected tracker store: "
+                f"{', '.join(sorted(missing_ids_in_tracker_store))}"
+            )
+
+        conversation_ids_to_process = conversation_ids_in_tracker_store & set(
+            self.requested_conversation_ids
+        )
+
+        if not conversation_ids_to_process:
+            raise NoEventsToMigrate(
+                "Could not find an overlap between the requested "
+                "conversation IDs and those found in the tracker store. Exiting."
+            )
+
+        return conversation_ids_to_process
+
+    def _fetch_events_within_time_range(self) -> List[Dict[Text, Any]]:
+        """Fetch all events for `conversation_ids` within the supplied time range.
+
+        Returns:
+            List of serialized events with added `sender_id` field.
+
+        """
+        conversation_ids_to_process = self.get_conversation_ids_to_process()
+
+        cli_utils.print_info(
+            f"Fetching events for {len(conversation_ids_to_process)} "
+            f"conversation IDs:"
+        )
+
+        events = []
+
+        for conversation_id in tqdm(conversation_ids_to_process, "conversation IDs"):
+            tracker = self.tracker_store.retrieve(conversation_id)
+            if not tracker:
+                logger.info(
+                    f"Could not retrieve tracker for conversation ID "
+                    f"'{conversation_id}'. Skipping."
+                )
+                continue
+
+            _events = tracker.current_state(EventVerbosity.ALL)["events"]
+
+            if not _events:
+                logger.info(
+                    f"No events to migrate for conversation ID '{conversation_id}'."
+                )
+                continue
+
+            # the conversation IDs are needed in the event publishing
+            for event in _events:
+                event["sender_id"] = conversation_id
+                events.append(event)
+
+        return self._sort_and_select_events_by_timestamp(events)
+
+    def _sort_and_select_events_by_timestamp(
+        self, events: List[Dict[Text, Any]]
+    ) -> List[Dict[Text, Any]]:
+        """Sort list of events by ascending timestamp, and select events within time range.
+
+        Raises `NoEventsInTimeRange` error if no events are found within the requested
+        time range.
+
+        Args:
+            events: List of serialized events to be sorted and selected from.
+
+        Returns:
+            List of serialized and sorted (by timestamp) events within the requested time
```

Comment:
optional: don't repeat the type 
```suggestion
            Serialized and sorted (by timestamp) events within the requested time range.
```